In [1]:
%load_ext autoreload
%autoreload

import sys

sys.path.append('../../utils')
from utils import PCA_reduction, PCA_components, PCA_snapshot

### PCA

In [2]:
PCA_reduction('../../../dumps/various_sizes/16K.csv','tree')

Variance      Training acc    Test acc  Components      Time (s)
----------  --------------  ----------  ------------  ----------
no pca            0.992049    0.988155  all             0.237223
1                 0.991971    0.983791  119             1.17098
0.99              0.991113    0.984726  102             0.97206
0.95              0.991113    0.984102  82              0.799969
0.9               0.991035    0.983791  64              0.682058
0.85              0.991113    0.985349  52              0.568896


In this case the accuracies are really similar, even slightly better, but we managed to reduced the number of features from 119 to 42, so more than 60%. The time as also be improved by half !

In [ ]:
PCA_snapshot('../../../dumps/time_analysis/threshold_3/3_20190615_31000.csv','tree','../../')